### PR TITLE
[Finder] Simplify SortableIterator

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

Spotted while working on #42106. The sort property is either a callable or zero. This means that `SortableIterator::getIterator()` currently handles impossible cases.

### CODE REVIEW

File: src/Symfony/Component/Finder/Iterator/SortableIterator.php [hunk 1]
```diff
@@ -83,15 +89,8 @@ public function __construct(\Traversable $iterator, $sort, bool $reverseOrder =
      */
     public function getIterator()
     {
-        if (1 === $this->sort) {
-            return $this->iterator;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Finder/Iterator/SortableIterator.php:87]: this early return prevented the `iterator_to_array` to array below, saving some memory in the process

File: src/Symfony/Component/Finder/Iterator/SortableIterator.php [hunk 2]
```diff
@@ -70,9 +74,11 @@ public function __construct(\Traversable $iterator, $sort, bool $reverseOrder =
                 return $order * ($a->getMTime() - $b->getMTime());
             };
         } elseif (self::SORT_BY_NONE === $sort) {
-            $this->sort = $order;
+            $this->sort = null;
         } elseif (\is_callable($sort)) {
-            $this->sort = $reverseOrder ? static function ($a, $b) use ($sort) { return -$sort($a, $b); } : $sort;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Finder/Iterator/SortableIterator.php:75]: this adds a call that the previous logic skipped

File: src/Symfony/Component/Finder/Iterator/SortableIterator.php [hunk 3]
```diff
@@ -70,9 +74,11 @@ public function __construct(\Traversable $iterator, $sort, bool $reverseOrder =
                 return $order * ($a->getMTime() - $b->getMTime());
             };
         } elseif (self::SORT_BY_NONE === $sort) {
-            $this->sort = $order;
+            $this->sort = null;
```
Comment (stof) [inline src/Symfony/Component/Finder/Iterator/SortableIterator.php:77]: This does not preserve the order, that was stored before.

File: src/Symfony/Component/Finder/Iterator/SortableIterator.php [hunk 4]
```diff
@@ -27,6 +27,10 @@ class SortableIterator implements \IteratorAggregate
     public const SORT_BY_NAME_NATURAL = 6;
 
     private $iterator;
+
+    /**
+     * @var \Closure|null
```
Comment (stof) [inline src/Symfony/Component/Finder/Iterator/SortableIterator.php:32]: The existing type is `\Closure|int|null` (`\Closure|1|-1|null` to be precise with literal types)

File: src/Symfony/Component/Finder/Iterator/SortableIterator.php (hunk 3 above)
Comment (derrabus) [reply src/Symfony/Component/Finder/Iterator/SortableIterator.php:77]: Right. I was probably a bit too eager here.

**Comment by derrabus** [comment]
I think, I misread the code while fixing the types. I'm closing this PR and will fix my PR against 6.0 accordingly.

**Pull request closed without merging**